src/libraries/Microsoft.PowerFx.Connectors/OpenApiExtensions.cs

```diff
@@ -0,0 +1,235 @@
+                }
+            }
+
+            return null;
+        }
+
+        public static bool IsTrigger(this OpenApiOperation op)
+        {
+            var isTrigger = op.Extensions.ContainsKey("x-ms-trigger");
+            return isTrigger;
+        }
+
+        public static bool TryGetDefaultValue(this OpenApiParameter param, out string defaultValue)
+        {
+            var x = param.Schema.Default;
+            if (x == null)
+            {
+                defaultValue = null;
+                return false;
+            }
+
+            if (x is OpenApiString str)
+            {
+                defaultValue = str.Value;
+                return true;
+            }
+
+            if (x is OpenApiInteger intVal)
+            {
+                defaultValue = intVal.Value.ToString();
+                return true;
+            }
+
+            if (x is OpenApiDouble dbl)
+            {
+                defaultValue = dbl.Value.ToString();
+                return true;
+            }
+
+            if (x is OpenApiBoolean b)
+            {
+                defaultValue = b.Value.ToString();
+                return true;
+            }
+
+            throw new NotImplementedException($"Unknown default value type {x.GetType().FullName}");
+        }
+
+        public static bool HasDefaultValue(this OpenApiParameter param)
+        {
+            return param.Schema.Default != null;
+        }
+
+        // Internal parameters are not showen to the user. 
+        // They can have a default value or be special cased by the infrastructure (like "connectionId").
+        public static bool IsInternal(this OpenApiParameter param) => param.Extensions.TryGetValue("x-ms-visibility", out _);
+
+        // See https://swagger.io/docs/specification/data-models/data-types/
+        public static FormulaType ToFormulaType(this OpenApiSchema schema)
+        {
+            switch (schema.Type)
+            {
+                case "string": return FormulaType.String;
+                case "number": return FormulaType.Number;
+                case "boolean": return FormulaType.Boolean;
+                case "integer": return FormulaType.Number;
+                case "array":
+                    var elementType = schema.Items.ToFormulaType();
+                    if (elementType is RecordType r)
+                    {
+                        return r.ToTable();
+                    }
+                    else if (elementType is not AggregateType)
+                    {
+                        // Primitives get marshalled as a SingleColumn table.
+                        // Make sure this is consistent with invoker. 
+                        var r2 = new RecordType().Add(TableValue.ValueName, elementType);
+                        return r2.ToTable();
+                    }
+                    else
+                    {
+                        throw new NotImplementedException();                    
+                    }
```

Comment:
Swagger supports mixed type arrays, should map to UntypedObject in the future.

Reply:
good point.